feat: Improve Rev Analytics onboarding

rafaeelaudibert opened the pull request:
Easy win, just display a banner saying we're syncing data warehouse data and it should be available pretty soon.

Closes #31260

rafaeelaudibert committed:
feat: Improve Rev Analytics onboarding

Easy win, just display a banner saying we're syncing data warehouse data and it should be available pretty soon.

rafaeelaudibert commented on frontend/src/lib/lemon-ui/LemonButton/LemonButton.scss:
```diff
@@ -46,7 +46,7 @@
     --lemon-button-transition: opacity 200ms ease, transform 200ms ease;
     --lemon-button-border-width: 0;
-    --lemon-button-bg-color: transparent;
+    --lemon-button-bg-color: var(--primary-3000-button-bg);
```
https://posthog.slack.com/archives/C025W6MN8SG/p1747855213733309

rafaeelaudibert replied:
updated it to target buttons inside `LemonBanner` only, slightly safer

rafaeelaudibert commented on products/revenue_analytics/frontend/RevenueAnalyticsScene.tsx:
```diff
@@ -165,20 +119,81 @@ export function RevenueAnalyticsSceneContent(): JSX.Element {
-                <RevenueAnalyticsFilters />
-            </div>
 
+            {isAnyRevenueSourceRunningForTheFirstTime && (
+                <LemonBanner
+                    type="success"
+                    className="mb-2"
+                    action={{ children: 'Refresh', onClick: () => window.location.reload() }}
+                >
+                    Some of your revenue data warehouse sources are currently still running for the first time. This
+                    means you might not see all of your revenue data yet.
+                    <br />
+                    We do, however, attempt to display partial data for these sources (e.g. the last 30 days of data).
+                    Refresh the page to see the latest data.
+                </LemonBanner>
+            )}
```
![image](https://github.com/user-attachments/assets/b831c859-bb69-43f7-8f99-ef08c2b972db)

rafaeelaudibert committed:
fix: Avoid bleeding LemonBanner background on LemonButton

rafaeelaudibert commented on products/revenue_analytics/frontend/revenueAnalyticsLogic.ts:
```diff
@@ -156,26 +153,33 @@ export const revenueAnalyticsLogic = kea<revenueAnalyticsLogicType>([
             ],
         ],
 
+        revenueEnabledEvents: [(s) => [s.events], (events) => events],
```
Because they're all revenueEnabled

rafaeelaudibert commented on products/revenue_analytics/frontend/RevenueAnalyticsScene.tsx:
```diff
@@ -93,67 +91,23 @@ export function RevenueAnalyticsScene(): JSX.Element {
+        return <RevenueAnalyticsSceneOnboarding />
     }
 
+    const isAnyRevenueSourceRunningForTheFirstTime =
+        revenueEnabledDataWarehouseSources?.some((source) => source.status === 'Running' && !source.last_run_at) ??
+        false
```
I AM using optional chaining :)

lricoy reviewed (approved):
I think we should also change the text color on those buttons now to something other than white on the dark mode because the default for the solid buttons seems to be black on dark mode.

It is probably a case by case change.

update: Just saw the comment, probably outdated snapshots

rafaeelaudibert committed:
feat: Add stories for the new onboarding banner

rafaeelaudibert committed:
feat: Add cross-sell opportunity from DWH sources table

rafaeelaudibert commented:
@lricoy Yeah, that sucked, I've changed the code to not change nearly as many buttons like my previous changes did. Also added some new snapshots, mind re-reviewing?

lricoy reviewed (approved):
Played around a bit and looks good. Ship it!

ps: I don't have the Revenue warehouse tables locally, so keep that in mind

rafaeelaudibert commented on frontend/__snapshots__/scenes-app-revenue-analytics--revenue-analytics-dashboard-sync-in-progress--dark.png:
Ugh, kinda ugly, will try again

rafaeelaudibert committed:
ui: Update some CSS

Make these look slightly nicer on dark mode when on top of themed bg

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

github-actions[bot] committed:
Update UI snapshots for `chromium` (7)
